public/javascripts/items.js

```diff
@@ -10,25 +10,83 @@ const fetchParams = {
 const params = completed =>
   Object.assign( {}, fetchParams, { body: JSON.stringify({ completed }) })
 
+const params2 = title =>
+  Object.assign( {}, fetchParams, { body: JSON.stringify({ title }) })
+
+const params3 = description =>
+  Object.assign( {}, fetchParams, { body: JSON.stringify({ description }) })
+
 $(document).ready( () => {
-  $( '.title' ).click( event => {
+  $('.updateTitle').keypress( event => {
+    element = $( event.target )
+    const id = element.data( 'id' )
+
+    if( event.charCode === 13 ) {
+      let updatedTitle = element[0].value
+      fetch( `/items/${id}/edit`, params2( updatedTitle ))
+      .then( result => result.json() )
+      .then( json => {
```

Comment:
This is not the API method we were chatting about earlier, but this is an alternative that allows us to move in that direction (should we not choose to modify the backend now...):
```js
const checkJsonForSuccessField = json => {
  if( json.success ) {
    Promise.resolve( json )
  } else {
    Promise.reject( json.message )
  }
}
```
Now, we can drop this function in anywhere that the backend may return a `json.success` flag, so that the code we write does not have to duplicate this common error check, and can instead just write the error management code:
```js
  .then( result => result.json() )
  .then( checkJsonForSuccessField )
  .then( json => {
    // Do the stuff
  })
  .catch( error => {
    // Handle the error (maybe with an error message or some form of UI update)
  })
```